src/dotty/tools/dotc/sbt/ThunkHolder.scala

```diff
@@ -0,0 +1,59 @@
+package dotty.tools.dotc
+package sbt
+
+import scala.annotation.tailrec
+import scala.collection.mutable.ListBuffer
+import xsbti.api
+
+/** Create and hold thunks */
```

Comment:
A thunk is an unevaluated value that can be evaluated once, it's a well known term (especially used in Haskell, but not only). I can make the description longer to make this clearer.

This trait was intentionally split from `ExtractAPICollector` and I'd like to keep it this way:
- It has private mutable state that does not need to be modified by `ExtractAPICollector`
- The implementation is somewhat tricky, but it can be understood on its own without worrying about how it's used in `ExtractAPICollector`
- Similarly, someone can understand how `ExtractAPICollector` works without having to know the implementation details of `ThunkHolder`